src/Cursor.js

```diff
@@ -37,8 +37,10 @@ function update(cmp, path, operation, nextValue) {
   var nextState;
 
   if (path.length > 0) {
+    var q = cmp._reactInternalInstance._pendingStateQueue;
```

Comment:
You can pass a function to setState now. I'm not sure how that helps this scenario but something to think about.